installer/environment.py

```
@@ -70,77 +109,94 @@
+            self.setup_authorized_keys()
+            local_ip = self.docker_setup.get_public_ip()
 
-        private_key_path, public_key_path = self.generate_ssh_key()
-        self.setup_authorized_keys()
-        local_ip = self.docker_setup.get_local_ip()
+            docker_context = self.docker_setup.setup()
 
-        docker_context = self.docker_setup.setup()
+            domain_not_provided = self.domains is None
+            if not domain_not_provided:
+                if not all(isinstance(domain, str) and '.' in domain for domain in self.domains.values()):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)